nixos/modules/services/web-apps/c2FmZQ-server.nix

```
@@ -0,0 +1,138 @@
+{ lib, pkgs, config, ... }:
+
+let
+  inherit (lib) mkEnableOption mkPackageOption mkOption types;
+
+  cfg = config.services.c2FmZQ-server;
+
+  argsFormat = with lib; {
+    type = with types; let
+      valueType = nullOr (oneOf [
+        bool int str
+      ]) // {
+        description = "c2FmZQ-server params";
+      };
+    in valueType;
+    generate = name: value:
+      let mkParam = k: v:
+            if v == null then []
+            else if isBool v then optional v ("--"+k)
+            else [("--"+k) v];
+      in escapeShellArgs (builtins.concatLists (mapAttrsToList mkParam value));
+  };
+in {
+  options.services.c2FmZQ-server = {
+    enable = mkEnableOption "c2FmZQ-server";
+
+    bindIP = mkOption {
+      type = types.str;
+      default = "127.0.0.1";
+      description =  "The local address to use.";
+    };
+
+    port = mkOption {
+      type = types.port;
+      default = 8080;
+      description =  "The local port to use.";
+    };
+
+    passphraseFile = mkOption {
+      type = types.str;
+      example = "/run/secrets/c2FmZQ/pwfile";
+      description = "Path to file containing the database passphrase";
+    };
+
+    package = mkPackageOption pkgs "c2FmZQ" { };
+
+    settings = mkOption {
+      type = types.submodule {
+        freeformType = argsFormat.type;
+
+        options = {
+          address = mkOption {
+            internal = true;
+            type = types.str;
+            default = "${cfg.bindIP}:${toString cfg.port}";
+          };
+
+          database = mkOption {
+            type = types.str;
+            default = "%S/c2FmZQ-server/data";
+            description = "Path of the database";
+          };
+
+          verbose = mkOption {
+            type = types.enum [ 1 2 3 ];
+            default = 2;
+            description = "The level of logging verbosity: 1:Error 2:Info 3:Debug";
+          };
+        };
+      };
+      description = ''
+        Configuration for c2FmZQ-server passed as CLI arguments.
+        Run `c2FmZQ-server help` for supported values.
+      '';
+      example = lib.literalExpression ''
+        {
+          verbose = 3;
+          allow-new-accounts = true;
+          auto-approve-new-accounts = true;
+          encrypt-metadata = true;
+          enable-webapp = true;
+        }
+      '';
```

Does this need to be a literalExpression? I thought not doing it would yield the same result?

```suggestion
      example = {
        verbose = 3;
        allow-new-accounts = true;
        auto-approve-new-accounts = true;
        encrypt-metadata = true;
        enable-webapp = true;
      };
```